main.nf

```diff
@@ -2,107 +2,228 @@
+}
+
+
+// Genome 1
+fasta1 = params.genome1 ? params.genomes[ params.genome1 ].fasta ?: false : false
+bt1 = params.genome1 ? params.genomes[ params.genome1 ].bowtie2 ?: false : false
+if ( params.fasta1 ){
+    fasta1 = file(params.fasta1)
+} else if (fasta1) {
+    fasta1 = file(params.genomes[ params.genome1 ].fasta, checkIfExists: true)
+}
+if( !fasta1.exists() ) exit 1, "Fasta1 file not found: ${params.fasta2}"
+
+if (params.index1) {
+    bt1 = params.index1
+}
+if (bt1) {
+    lastPath = bt1.lastIndexOf(File.separator)
+    bt1_dir = bt1.substring(0, lastPath+1)
+    bt1_index = bt1.substring(lastPath+1)
+
+    Channel
+        .fromPath(bt1_dir+"/*.bt2")
+        .ifEmpty {exit 1, "Cannot find any index matching : ${bt1}\n"}
+        .collect()
+        .set {bt1_ch}
+} else {
+    bt1_index = params.name1
+}
+
+
+// Genome 2
+fasta2 = params.genome2 ? params.genomes[ params.genome2 ].fasta ?: false : false
+bt2 = params.genome2 ? params.genomes[ params.genome2 ].bowtie2 ?: false : false
+if ( params.fasta2 ){
+    fasta2 = file(params.fasta2)
+} else if (fasta2) {
+    fasta2 = file(params.genomes[ params.genome2 ].fasta, checkIfExists: true)
+}
+if( !fasta2.exists() ) exit 1, "Fasta2 file not found: ${params.fasta2}"
+
+if (params.index2) {
+    bt2 = params.index2
+}
+
+if (bt2) {
+    lastPath = bt2.lastIndexOf(File.separator)
+    bt2_dir = bt2.substring(0, lastPath+1)
+    bt2_index = bt2.substring(lastPath+1)
+
+    Channel
+        .fromPath(bt2_dir+"/*.bt2")
+        .ifEmpty {exit 1, "Cannot find any index matching : ${params.bt2}\n"}
+        .collect()
+        .set {bt2_ch}
+
+} else {
+    bt2_index = params.name2
+}
+
+// Genome 3
+if (params.name3) {
+    fasta3 = params.genome3 ? params.genomes[ params.genome3 ].fasta ?: false : false
+    bt3 = params.genome3 ? params.genomes[ params.genome3 ].bowtie2 ?: false : false
+    if ( params.fasta3 ){
+        fasta3 = file(params.fasta3)
+    } else if (fasta3) {
+        fasta3 = file(params.genomes[ params.genome3 ].fasta, checkIfExists: true)
+    }
+    if( !fasta3.exists() ) exit 1, "Fasta3 file not found: ${params.fasta3}"
+
+    if (params.index3) {
+        bt3 = params.index3
+    }
+    if (bt3) {
+        lastPath = bt3.lastIndexOf(File.separator)
+        bt3_dir = bt3.substring(0, lastPath+1)
+        bt3_index = bt3.substring(lastPath+1)
+
+        Channel
+            .fromPath(bt3_dir+"/*.bt2")
+            .ifEmpty {exit 1, "Cannot find any index matching : ${params.bt3}\n"}
+            .collect()
+            .set {bt3_ch}
+    } else {
+        bt3_index = params.name3
+    }
+    
+}
+
+
+// Has the run name been specified by the user?
+//  this has the bonus effect of catching both -name and --name
+custom_runName = params.name
+if( !(workflow.runName ==~ /[a-z]+_[a-z]+/) ){
+  custom_runName = workflow.runName
+}
+
+
+// Stage config files
+ch_multiqc_config = Channel.fromPath(params.multiqc_config)
```

Comment:
I guess these should now be changed to `file(params.multiqc_config, checkIfExists: true)`. Need to update the template!

Reply:
Fixed with #7

Reply:
`ch_output_docs` needs to be changed too. Everywhere you are creating a file channel that isnt a glob should use `file(params.test, checkIfExists: true)`. Easier to search for `fromPath` and replace where necessary.